app/assets/javascripts/proposals.coffee

```diff
@@ -9,6 +9,17 @@ $(document).ready ->
       # validate or revalidate on close
       $(this).valid()
 
+    # The jquery validation below is not sufficient to drive the error displays
+    # so changing the note should trigger a class change on the reasons select2
+    $('#rejection_note').change ->
+      if $('#rejection_note').val()
```

Comment:
so calling `updateRequiredIcons()` is not sufficient, you need to add or remove the class on the other field for this to work?

Reply:
Correct.